js/all-the-ways-to-color-your-text-in-css/all-the-ways-to-color-your-text-in-css.md

```diff
@@ -0,0 +1,178 @@
+
+```css
+// color value with no alpha or fully opaque
+h1 {
+    color: #FF0000;
+}
+```
+
+> <h3 style="color: #ff0000">I am completely opaque<h3>
+
+```css
+// semi-transparent color
+h1 {
+    color: #FF000080;
+}
+```
+
+> <h3 style="color: #FF000080">I am semi-transparent<>
+
+📌 **Browser Compatibility**
+
+The browser support for #RRGGBBAA hex codes is great. IE doesn’t support it but otherwise you should be covered. Check [Can I Use](https://caniuse.com/#feat=css-rrggbbaa) for details if you’re curious.
+
+#### 3. RGB Values
+
+An RGB color values is specified with the [[rgb]] function which has the following syntax:
+
+> rgb (red, green, blue)
+
+The value for each color in the function defines the intensity of the color. Each color value is an integer between [[0 and 255]] or a percentage value from [[0% to 100%]].
+
+For example, these all are valid rgb color values
+
+```css
+// valid rgb values
+
+rgb (0, 0, 255)
+rgb (0%, 0%, 100%)
+rgb (0.5%, 3.5%, 0.5%)
+rgb (89, 55, 255)
+```
+
+📌 **RGBA**
+
+RGBA color values are an extension of RGB color values with an alpha value which provides opacity to the color similar to **#RRGGBBAA**. An RGBA color is specified with the [[rgba()]] function, which has the following syntax
+
+> rgba (red, green, blue, alpha)
+
+The alpha parameter in the above function is a number between 0.0 (fully transparent) and 1.0 (fully opaque).
+
+```css
+// rgba value to provide full opacity
+h1 {
+    color: rgba (0, 0, 255, 1);
+}
+```
+
+> <h3 style="color: rgba(0, 0, 255, 1)">I am fully opaque<h3>
+
+```css
+// rgba value with transparency
+h1 {
+    color: rgba (0, 0, 255, 0.3);
+}
+```
+
+> <h3 style="color: rgba(0, 0, 255, 0.3)">I am semi-transparent</h3>
+
+📌 **Browser Compatibility**
+
+The browser support for rgba() is great. Check [Can I Use](https://caniuse.com/#feat=rgba) for details if you’re curious.
+
+#### 4. HSL Values
+
+HSL stands for hue, saturation, and lightness - and represents a cylindrical-coordinate representation of colors. This looks syntactically similar to rgb values but the ranges are different. An HSL color value is specified with the hsl() function, which has the following syntax:
+
+> hsl (hue, saturation, lightness)
+
+**Hue**: Think of a color wheel. Around 0 and 360 degrees are reds. 120 degree is where greens are and 240 degree are blues. Use anything in between 0-360. Values above and below will be modulus 360.
+**Saturation**: 0% is completely desaturated (grayscale). 100% is fully saturated (full color).
+**Lightness**: 0% is completely dark (black). 100% is completely light (white). 50% is average lightness.
+**alpha**: Opacity/Transparency value. 0 is fully transparent. 1 is fully opaque. 0.5 is 50% transparent.
```

Comment:
since alpha is defined in hsla you should make some changes. Either describe hsla here and quickly mention that hsl exists too. Because people would already get the hang of rgb + rgba thing in previous section